src/lib.rs

```diff
@@ -0,0 +1,437 @@
+        subject_unique_id: None,
+        extensions: Some(extensions),
+    }
+    .sign(pki)
+    .or(Err(StatusCode::INTERNAL_SERVER_ERROR))
+}
+
+/// Receives:
+/// ASN.1 SEQUENCE OF CertRequest.
+/// Returns:
+/// ASN.1 SEQUENCE OF Output.
+pub async fn attest(
+    TypedHeader(ct): TypedHeader<ContentType>,
+    body: Bytes,
+    Extension(state): Extension<Arc<State>>,
+) -> Result<Vec<u8>, impl IntoResponse> {
+    // Decode the signing certificate and key.
+    let issuer = Certificate::from_der(&state.crt).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
+    let isskey = PrivateKeyInfo::from_der(&state.key).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
+
+    const TTL: Duration = Duration::from_secs(60 * 60 * 24 * 28);
+    let now = SystemTime::now();
+    let end = now + TTL;
+    let validity = Validity {
+        not_before: Time::try_from(now).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?,
+        not_after: Time::try_from(end).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?,
+    };
+
+    // Check for correct mime type.
+    let reqs = match ct.to_string().as_ref() {
+        PKCS10 => vec![CertReq::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?],
+        BUNDLE => Vec::from_der(body.as_ref()).or(Err(StatusCode::BAD_REQUEST))?,
+        _ => return Err(StatusCode::BAD_REQUEST),
+    };
+
+    // Decode and verify the certification requests.
+    reqs.into_iter()
+        .map(|cr| {
+            // Create the basic subject alt name.
+            let name = Ia5StringRef::new("foo.bar.hub.profian.com")
+                .or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
+            let mut sans = vec![GeneralName::DnsName(name)];
+
+            // Optionally, add the configured subject alt name.
+            if let Some(name) = &state.san {
+                let name = Ia5StringRef::new(name).or(Err(StatusCode::INTERNAL_SERVER_ERROR))?;
+                sans.push(GeneralName::DnsName(name));
+            }
+            attest_request(
+                &issuer,
+                &isskey,
+                SubjectAltName(sans),
+                cr,
+                &validity,
+                &state,
+            )
+        })
+        .collect::<Result<Vec<_>, _>>()
+        .and_then(|issued| {
+            let issued: Vec<Certificate<'_>> = issued
+                .iter()
+                .map(|c| Certificate::from_der(c).or(Err(StatusCode::INTERNAL_SERVER_ERROR)))
+                .collect::<Result<_, _>>()?;
+
+            match ct.to_string().as_ref() {
+                PKCS10 => vec![issuer, issued[0].clone()].to_vec(),
+                BUNDLE => Output {
+                    chain: vec![issuer],
+                    issued,
+                }
+                .to_vec(),
+                _ => return Err(StatusCode::BAD_REQUEST),
+            }
+            .or(Err(StatusCode::INTERNAL_SERVER_ERROR))
+        })
+}
+
+pub fn init_tracing() {
+    TRACING.call_once(|| {
```

Comment:
In production this will only ever be called once, `TRACING` only makes sense for tests.

If you want to reuse this logic, then remove `TRACING.call_once` from this function and instead, refactor unit tests to call `TRACING.call_once(init_tracing)`

Reply:
Fixed